applications/acdc/src/acdc_queue_manager.erl

```diff
@@ -947,4 +958,47 @@ update_properties(QueueJObj, State) ->
     State#state{
       enter_when_empty=kz_json:is_true(<<"enter_when_empty">>, QueueJObj, 'true')
                ,moh=kz_json:get_ne_value(<<"moh">>, QueueJObj)
+               ,announcements_config=announcements_config(QueueJObj)
      }.
+
+-spec announcements_config(kz_json:object()) -> kz_proplist().
+announcements_config(Config) ->
+    kz_json:recursive_to_proplist(
+      kz_json:get_json_value(<<"announcements">>, Config, kz_json:new())).
+
+-spec cancel_position_announcements(kapps_call:call() | 'false', map()) ->
+                                           map().
+cancel_position_announcements('false', Pids) -> Pids;
+cancel_position_announcements(Call, Pids) ->
+    CallId = kapps_call:call_id(Call),
+    case maps:is_key(CallId, Pids) of
+        'true' ->
+            lager:debug("cancelling announcements for ~s", [CallId]),
+            {Pid, Pids1} = maps:take(CallId, Pids),
```

Comment:
If using maps:take here you don't really need to check maps:is_key beforehand (you can move the take into that case statement and handle either the return or error return).

Also as you may be aware, maps:take will NOT work on Erlang 18, and afaik Kazoo has not moved to 19 yet.

Reply:
We should move soon.

Reply:
"Kazoo 4 targets Erlang 19+" - from https://github.com/2600hz/kazoo/blob/master/doc/installation.md

Reply:
Either way, I've moved to using get/2 and remove/2 which are compatible with 18 through 20 :)

Reply:
@kalda341 That was merged soon after my comment ;)